tests/api/tests.py

```diff
@@ -0,0 +1,381 @@
+                else:
+                    raise Exception('Bad method name')
+
+        self._send_response(status_code, data)
+
+    def do_GET(self):
+        data = ''
+        status_code = requests.codes.not_found
+
+        if self._match(self.re_products, self.path):
+            if 'short=fedora' in str(self.match.group(1)):
+                data = _paged_results([self.products['fedora']])
+            else:
+                data = _paged_results(self.products)
+            status_code = requests.codes.ok
+        elif self._match(self.re_product, self.path):
+            product_id = self.match.group(1)
+            data = self.products.get(product_id)
+            if data:
+                status_code = requests.codes.ok
+        elif self._match(self.re_token, self.path):
+            data = {'token': '1'}
+            status_code = requests.codes.ok
+
+        self._send_response(status_code, data)
+
+    def do_POST(self):
+        data = ''
+        status_code = requests.codes.not_found
+
+        if self._match(self.re_cpes, self.path):
+            data = self._data()
+            pk = self._find_available_pk(self.cpes)
+            data['id'] = pk
+            self.cpes[pk] = data
+            status_code = requests.codes.ok
+        else:
+            status_code = requests.codes.unauthorized
+
+        self._send_response(status_code, data)
+
+    def do_PATCH(self):
+        self._do('PATCH')
+
+    def do_PUT(self):
+        self._do('PUT')
+
+    def do_DELETE(self):
+        self._do('DELETE')
+
+    def log_message(self, format, *args):
+        """
+        Omit printing on console.
+        """
+        pass
+
+    # Workaround for 'Connection reset by peer' errors in Python 2.6.
+    def handle(self):
+        time.sleep(0.001)
+        super(_MockPDCServerRequestHandler, self).handle()
+
+
+class PDCClientTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        port_value = os.getenv(PORT_ENV_VAR_NAME)
+        cls.port = int(port_value) if port_value else DEFAULT_PORT
+
+        server_name_value = os.getenv(SERVER_ENV_VAR_NAME)
+        cls.server_name = server_name_value if server_name_value else DEFAULT_SERVER
+
+        cls.server = HTTPServer((cls.server_name, cls.port), _MockPDCServerRequestHandler)
+
+        cls.server_thread = Thread(target=cls.server.serve_forever)
+        cls.server_thread.setDaemon(True)
+        cls.server_thread.start()
+
+        cls.url = 'http://{server}:{port}{api_path}'.format(
+            server=cls.server_name,
+            port=cls.port,
+            api_path=API_PATH,
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def setUp(self):
+        # Python 2.6 compatibility
+        if not hasattr(PDCClientTestCase, 'url'):
+            PDCClientTestCase.setUpClass()
+
+        self.client = PDCClient(
+            server=self.url,
+            ssl_verify=False,
+        )
+
+    def test_request_response(self):
```

Comment:
I don't quite understand this case. Is this testing the Mock server setup correctly? I think other cases already cover this.

Reply:
OK, I'll remove it. It was the first test just to verify the server is running.